metadata-io/src/test/java/com/linkedin/metadata/entity/DatastaxEntityServiceTest.java

```diff
@@ -0,0 +1,739 @@
+package com.linkedin.metadata.entity;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.Session;
+import com.google.common.base.Equivalence;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.CorpuserUrn;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.data.ByteString;
+import com.linkedin.data.template.DataTemplateUtil;
+import com.linkedin.data.template.JacksonDataTemplateCodec;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.dataset.DatasetProfile;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.identity.CorpUserInfo;
+import com.linkedin.metadata.query.ListUrnsResult;
+import com.linkedin.metadata.utils.EntityKeyUtils;
+import com.linkedin.metadata.utils.PegasusUtils;
+import com.linkedin.metadata.aspect.Aspect;
+import com.linkedin.metadata.aspect.CorpUserAspect;
+import com.linkedin.metadata.aspect.CorpUserAspectArray;
+import com.linkedin.metadata.aspect.VersionedAspect;
+import com.linkedin.metadata.entity.datastax.DatastaxAspect;
+import com.linkedin.metadata.entity.datastax.DatastaxAspectDao;
+import com.linkedin.metadata.entity.datastax.DatastaxEntityService;
+import com.linkedin.metadata.event.EntityEventProducer;
+import com.linkedin.metadata.key.CorpUserKey;
+import com.linkedin.metadata.models.registry.ConfigEntityRegistry;
+import com.linkedin.metadata.models.registry.EntityRegistry;
+import com.linkedin.metadata.models.registry.MergedEntityRegistry;
+import com.linkedin.metadata.run.AspectRowSummary;
+import com.linkedin.metadata.snapshot.CorpUserSnapshot;
+import com.linkedin.metadata.snapshot.Snapshot;
+import com.linkedin.mxe.GenericAspect;
+import com.linkedin.mxe.MetadataAuditOperation;
+import com.linkedin.mxe.MetadataChangeProposal;
+import com.linkedin.mxe.SystemMetadata;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nonnull;
+import org.mockito.Mockito;
+import org.testcontainers.containers.CassandraContainer;
+import org.testcontainers.utility.DockerImageName;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+
+public class DatastaxEntityServiceTest {
```

Comment:
This file is very similar to `EbeanEntityServiceTest`. The tests and common helper methods should be moved into a base test class and only implementation specific tests and code should be in these test files.